expression/builtin_time_vec.go

```diff
@@ -2702,11 +2702,95 @@ func (b *builtinUTCTimestampWithoutArgSig) vecEvalTime(input *chunk.Chunk, resul
 }
 
 func (b *builtinConvertTzSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinConvertTzSig) vecEvalTime(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+
+	// fromTz
+	fromTzBuf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(fromTzBuf)
+	errFromTz := b.args[1].VecEvalString(b.ctx, input, fromTzBuf)
+
+	// toTz
+	toTzBuf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(toTzBuf)
+	errToTz := b.args[2].VecEvalString(b.ctx, input, toTzBuf)
+
+	// dt
+	errDt := b.args[0].VecEvalTime(b.ctx, input, result)
+
+	result.MergeNulls(fromTzBuf, toTzBuf)
+	times := result.Times()
+	fsp := int8(b.tp.Decimal)
+	var fromTzMatched, toTzMatched bool
+	var fromTzStr, toTzStr string
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+
+		fromTzStr = fromTzBuf.GetString(i)
+		toTzStr = toTzBuf.GetString(i)
+		if errFromTz != nil || errToTz != nil || errDt != nil ||
```

Comment:
`if errFromTz != nil || errToTz != nil || errDt != nil {`
we can use `result.SetNulls(0, n, true)`

Reply:
Fixed.